### PR TITLE
cpu/stm32f1: optimized baudrate calculation

### PR DESCRIPTION
**Pull request by haukepetersen**
I thought I did optimize this quite a while ago - seems like I didn't :-)

So no need to use floats here - same results saving ~500 byte...

### CODE REVIEW

File: cpu/stm32f1/periph/uart.c [hunk 1]
```diff
@@ -136,9 +135,9 @@ int uart_init_blocking(uart_t uart, uint32_t baudrate)
     gpio_init(rx_pin, GPIO_DIR_IN, GPIO_NOPULL);
 
     /* configure UART to mode 8N1 with given baudrate */
-    divider = ((float)bus_freq) / (16 * baudrate);
-    mantissa = (uint16_t)floorf(divider);
-    fraction = (uint8_t)floorf((divider - mantissa) * 16);
+    bus_freq /= baudrate;
```
Comment (PeterKietzmann) [inline cpu/stm32f1/periph/uart.c]: `/=`? Is `bus_freq`actually initialized somewhere before?

**Comment by PeterKietzmann** [comment]
@haukepetersen I'll try to test these PRs in the next week once I can run them on the regarding hardware

File: cpu/stm32f1/periph/uart.c (hunk 1 above)
Comment (OlegHahm) [inline cpu/stm32f1/periph/uart.c]: Yes, look at the switch-case statement. If no proper UART is given it will return.

Comment (PeterKietzmann) [inline cpu/stm32f1/periph/uart.c]: whoops, was blind that morning.

**Comment by PeterKietzmann** [comment]
ACK

**Comment by PeterKietzmann** [comment]
Hi experts! Any ideas on how to interpret the build test result? All groups successfully passed but the overall test says "failed"

**Comment by OlegHahm** [comment]
No idea. Should we just kick it?

**Comment by PeterKietzmann** [comment]
"Kick it" means merge? If so I'd say yes.

**Comment by thomaseichinger** [comment]
@PeterKietzmann I guess with "kick it" @OlegHahm meant to kick Travis. :wink:

**Comment by OlegHahm** [comment]
Yes, but I'm fine with Peter's interpretation, too. ;)

**Pull request merged**